rails/Dockerfile

```diff
@@ -1,74 +1,74 @@
+FROM docker.io/ruby:${RUBY_VERSION}-alpine3.12
+
+RUN apk add \
+  # The Ruby Gem "delayed-web" has a transitive depdency for "racc" which builds
+  # native extensions and requires a C toolchain (it attempts to use gcc):
+  # "Gem::Ext::BuildError: ERROR: Failed to build gem native extension."
+  # "The compiler failed to generate an executable file. (RuntimeError)"
+  # "You have to install development tools first."
+  build-base \
+  # The Ruby Gem "mimemagic" requires Freedesktop.org Shared MIME Info to be installed:
+  # "Could not find MIME type database in the following locations:"
+  # "Ensure you have either installed the shared-mime-info package for your distribution..."
+  shared-mime-info \
+  # The Ruby Gem "mysql2" requires a MySQL client library to link against
+  # See: https://github.com/brianmario/mysql2#general-instructions
+  mariadb-dev \
+  # The "execjs" requires a supported JavaScript runtime to be installed
+  # "/usr/local/bundle/gems/execjs-2.8.1/lib/execjs/runtimes.rb:58:in `autodetect':
+  # Could not find a JavaScript runtime. See https://github.com/rails/execjs for a list of available runtimes.
+  # (ExecJS::RuntimeUnavailable)"
+  # See: https://github.com/rails/execjs#execjs
+  nodejs \
+  # "/usr/local/bundle/gems/tzinfo-2.0.4/lib/tzinfo/data_source.rb:159:in
+  # `rescue in create_default_data_source': tzinfo-data is not present. Please add gem 'tzinfo-data' 
+  # to your Gemfile and run bundle install (TZInfo::DataSourceNotFound)"
+  tzdata \
+  # See: https://github.com/sstephenson/ruby-yui-compressor#label-Installing+and+loading+Ruby-YUI+Compressor
+  openjdk8-jre
+
+# If this is changed it also needs to be changed in unicorn.rb
+ENV APP_HOME /rigse/
+
+RUN adduser -D -u 13000 app
+
+RUN mkdir -v $APP_HOME
+RUN chown app $APP_HOME
 WORKDIR $APP_HOME
 
-ADD Gemfile* $APP_HOME/
-
-ENV BUNDLE_GEMFILE=$APP_HOME/Gemfile
-
-# need to copy the Gemfile.lock created during build so it isn't overriden by the following add
-RUN bundle install --without development test && \
-    cp Gemfile.lock Gemfile.lock-docker
-ADD . $APP_HOME
+USER app
 
-# get files into the right place
-RUN mv -f Gemfile.lock-docker Gemfile.lock && \
-    cp config/database.sample.yml config/database.yml && \
-    cp -r docker/prod/config/. config/
+# Copy Gemfile and its lock file
+COPY --chown=app ./Gemfile* $APP_HOME
+# TODO: bundle config set without 'development test'
+RUN bundle install --without development test
+COPY --chown=app ./ $APP_HOME
 
-## Configured nginx (after bundler so we don't have to wait for bundler to change config)
-RUN echo "\ndaemon off;" >> /etc/nginx/nginx.conf
-RUN chown -R www-data:www-data /var/lib/nginx
+RUN \
+  cp ./config/database.sample.yml ./config/database.yml \
+  && cp ./config/settings.sample.yml ./config/settings.yml \
+  && cp ./config/app_environment_variables.sample.rb app_environment_variables.rb \
+  && cp ./docker/prod/config/mailer.yml ./config/mailer.yml
```

Comment:
The old DockerFile was copying `./docker/prod/config` to `./config`. This new Dockerfile should probably do the same. The settings.sample.yml is not the same as `./docker/prod/config/settings.yml`.  And there are several other files in `./docker/prod/config` which do not seem to be copied by this new Dockerfile.